Packs/AzureResourceGraph/Integrations/AzureResourceGraph/AzureResourceGraph.py

```diff
@@ -0,0 +1,248 @@
+    try:
+        result = client.list_operations()
+        if result:
+            return 'ok'
+    except DemistoException as e:
+        return_error(f"Test connection failed with message {e}")
+
+
+## Helper Methods
+
+def pagination(response, page_size, page_number):
+    """Method to generate a page (slice) of data.
+    Args:
+        response: The response from the API.
+        limit: Maximum number of objects to retrieve.
+        page: Page number
+    Returns:
+        Return a list of objects from the response according to the page and limit per page.
+    """
+    if page_size > MAX_PAGE_SIZE:
+        page_size = MAX_PAGE_SIZE
+    
+    starting_index = (page_number - 1) * page_size
+    ending_index = starting_index + page_size
+    return response[starting_index:ending_index]
+
+
+def main():
+    params: dict = demisto.params()
+    args = demisto.args()
+    server = params.get('host', 'https://management.azure.com').rstrip('/')
+    tenant = params.get('cred_token', {}).get('password') or params.get('tenant_id')
+    auth_and_token_url = params.get('cred_auth_id', {}).get('password') or params.get('auth_id')
+    if not tenant or not auth_and_token_url:
+        return_error('Token and ID must be provided.')
+    enc_key = params.get('cred_enc_key', {}).get('password') or params.get('enc_key')
+    certificate_thumbprint = params.get('cred_certificate_thumbprint', {}).get(
+        'password') or params.get('certificate_thumbprint')
+    private_key = params.get('private_key')
+    verify = not params.get('unsecure', False)
+    subscription_id = args.get('subscription_id') or params.get(
+        'cred_subscription_id', {}).get('password') or params.get('subscription_id')
+    proxy: bool = params.get('proxy', False)
+    self_deployed: bool = params.get('self_deployed', False)
+    if not self_deployed and not enc_key:
+        raise DemistoException('Key must be provided. For further information see '
+                               'https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication')
+    elif not enc_key and not (certificate_thumbprint and private_key):
+        raise DemistoException('Key or Certificate Thumbprint and Private Key must be providedFor further information see '
+                               'https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication')
+    ok_codes = (200, 201, 202, 204)
+
+    commands_without_args: Dict[Any, Any] = {
+        'test-module': test_module
+    }
+
+    commands_with_args: Dict[Any, Any] = {
+        'azure-rg-query': query_resources_command,
+        'azure-rg-list-operations': list_operations_command
+    }
+
+    commands_with_args_and_params: Dict[Any, Any] = {
+    }
+
+    '''EXECUTION'''
+    command = demisto.command()
+    LOG(f'Command being called is {command}')
+
+    try:
+        # Initial setup
+        if not subscription_id:
+            return_error('A subscription ID must be provided.')
```

Comment:
Its better practice to take this logic outside of the main.

Reply:
Updated in dff3feb